### ISSUE TITLE
wan finetune

### ISSUE THREAD

**Issue by betterftr** [kohya-ss/musubi-tuner#542]
Nothing fancy just wanted finetuning for wan.. Might have made mistakes here and there so no pr
https://github.com/betterftr/musubi-tuner/tree/main
two files have been added  "src/musubi_tuner/wan_train.py" and "wan_train.py" wrapper

tested with
[general]
enable_bucket = true
bucket_no_upscale = false
caption_extension = ".txt"
batch_size = 8
resolution = [512, 512]

and working, samples ok..

![Image](https://github.com/user-attachments/assets/be816dfb-08f5-4a5f-a250-929b0c8e1702)

basically just mirrored qwen_image_train_network.py > qwen_image_train.py with its args like --fused_backward_pass --full_bf16 --mem_eff_save

full command I used:

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 wan_train.py --task t2v-A14B --dit C:/train/ckpts/wan22_t2v_14B_low_noise_bf16.safetensors --dataset_config C:/train/hunyuan_dataset.toml --sdpa --mixed_precision bf16 --optimizer_type adafactor --optimizer_args scale_parameter=False relative_step=False warmup_init=False --learning_rate 1e-5 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --timestep_sampling shift --discrete_flow_shift 3.0 --max_train_epochs 1000 --save_every_n_epochs 1 --save_state --seed 1234 --output_dir C:/train/saves --output_name lacey_512 --vae C:/train/ckpts/wan_2.1_vae.safetensors --t5 C:/train/ckpts/models_t5_umt5-xxl-enc-bf16.pth --sample_prompts "C:/train/sample/random_prompt.txt" --sample_every_n_epochs 1 --logging_dir "C:/train/logs" --lr_warmup_steps "500" --lr_scheduler "linear" --preserve_distribution_shape --min_timestep "0" --max_timestep "1000" --preserve_distribution_shape --blocks_to_swap 29 --fused_backward_pass --full_bf16 --mem_eff_save

**Comment by FurkanGozukara** [comment]
@kohya-ss any chance you can implement Wan fine tuning as well?

and merge this because we need it : https://github.com/kohya-ss/musubi-tuner/pull/585
